app/models/foreman_openscap/scap_content.rb

```diff
@@ -1,27 +1,25 @@
 require 'digest/sha2'
-require 'openscap/ds/sds'
-require 'openscap/source'
-require 'openscap/xccdf/benchmark'
 
 module ForemanOpenscap
   class DataStreamValidator < ActiveModel::Validator
     def validate(scap_content)
       return unless scap_content.scap_file_changed?
 
-      allowed_type = 'SCAP Source Datastream'
-      if scap_content.source.try(:type) != allowed_type
-        scap_content.errors[:base] << _("Uploaded file is not #{allowed_type}.")
+      if (proxy = SmartProxy.with_features('Openscap').first)
+        api = ProxyAPI::Openscap.new(:url => proxy.url)
+      else
+        scap_content.errors.add(:base, _('No Proxy with OpenScap features.'))
         return false
       end
 
-      begin
-        scap_content.source.validate!
-      rescue OpenSCAP::OpenSCAPError => e
-        scap_content.errors[:base] << e.message
+      errors = api.validate_scap_content(scap_content.scap_file)
```

Comment:
this can fail on timeout (and other network issues) so it should be wrapped, also in scenario where I have 2 proxies, first is failing (down) it can't use the second, would it make sense to iterate over all proxies with_features 'Openscap' and `break` with first success? sorry I noticed this during testing, it will need changing a bit also the `if (proxy = ...` that you already changed